### PR TITLE
add `_ctypes.pyi`

### PR DESCRIPTION
**Pull request by junkmd**
see #8571

### CODE REVIEW

File: stdlib/_ctypes.pyi [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+        hresult: int
+        text: str | None
+        details: _COMError_Details
+
+        def __init__(self, hresult: int, text: str | None, details: _COMError_Details) -> None: ...
+
+    def CopyComPointer(src: _PointerLike, dst: _PointerLike | _CArgObject) -> int:
+        """CopyComPointer(src, dst) -> HRESULT value"""
+        ...
```
Comment (hauntsaninja) [inline stdlib/_ctypes.pyi:21]:
```suggestion
    class COMError(Exception):
        hresult: int
        text: str | None
        details: _COMError_Details

        def __init__(self, hresult: int, text: str | None, details: _COMError_Details) -> None: ...

    def CopyComPointer(src: _PointerLike, dst: _PointerLike | _CArgObject) -> int: ...
```

File: stdlib/_ctypes.pyi [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    class COMError(Exception):
+        hresult: int
+        text: str | None
+        details: _COMError_Details
+
+        def __init__(self, hresult: int, text: str | None, details: _COMError_Details) -> None: ...
+
+    def CopyComPointer(src: _PointerLike, dst: _PointerLike | _CArgObject) -> int: ...
+
+def __getattr__(name: str) -> Any: ...  # incomplete
```
Comment (hauntsaninja) [inline stdlib/_ctypes.pyi:19]:
```suggestion
```

**Comment by junkmd** [comment]
@hauntsaninja 
Thank you for the code correction.

Currently, all tests are passed.

Is there anything else I should do?

**Review by AlexWaygood — COMMENTED** [review]
I know little about `ctypes`, but this all looks good as far as I can see. Can confirm that `COMError` takes exactly three arguments on my Windows machine, and `_COMError_Details` looks good from all the examples I can find online!

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**